onnx/checker.cc

```diff
@@ -226,6 +228,160 @@ void check_tensor(const TensorProto& tensor, const CheckerContext& ctx) {
 #undef check_field
 }
 
+// Check that the index data stored in a SparseTensorProto is valid.
+// indices: a 1-dimensional tensor; indices[i] represents the
+// linearized index value for the i-th nonzero value.
+void check_sparse_tensor_indices_1(
+    const TensorProto& indices,
+    const SparseTensorProto& sparse_tensor_proto,
+    size_t nnz) {
+  int dense_rank = sparse_tensor_proto.dims_size();
+  int64_t dense_size = 1;
+  for (int i = 0; i < dense_rank; ++i)
+    dense_size *= sparse_tensor_proto.dims(i);
+  if (static_cast<size_t>(indices.dims(0)) != nnz)
+    fail_check(
+        "Sparse tensor indices (",
+        indices.name(),
+        ") does not have NNZ values.");
```

Comment:
```suggestion
        ") does not have the right number of values.");
```
Please also print out the right value, `nnz`, and the existing value, `indices.dims(0)`.

Reply:
Thanks. I was puzzled about the original message … I guess I meant to write "does not have ", nnz, " values" !